### ISSUE TITLE
Error when running example code for GPFIFOSearcher

### ISSUE THREAD

**Issue by yx1215** [awslabs/autogluon#685]
When I tried to run the example code for searcher [here](https://autogluon.mxnet.io/api/autogluon.searcher.html), I will get the following error:

```
Traceback (most recent call last):
  File "/Users/yixiaxia/Desktop/test/test.py", line 17, in <module>
    searcher = GPFIFOSearcher(train_fn.cs)
TypeError: __init__() takes 1 positional argument but 2 were given
```

The following is the minimum script to reproduce the error:

```
import numpy as np
from autogluon.searcher import GPFIFOSearcher
import autogluon as ag


@ag.args(
    lr=ag.space.Real(1e-3, 1e-2, log=True),
    wd=ag.space.Real(1e-3, 1e-2))
def train_fn(args, reporter):
    print('lr: {}, wd: {}'.format(args.lr, args.wd))
    for e in range(10):
        dummy_accuracy = 1 - np.power(1.8, -np.random.uniform(e, 2 * e))
        reporter(epoch=e + 1, accuracy=dummy_accuracy, lr=args.lr, wd=args.wd)


print(train_fn.cs)
searcher = GPFIFOSearcher(train_fn.cs)
print(searcher.get_config())
```

Actually this is just the copy-paste of what we have on that website.

**Comment by jwmueller** [comment]
@yx1215 you can instead follow this tutorial: 
https://autogluon.mxnet.io/tutorials/course/mlp.html

Or use the following code instead:

```
import numpy as np
from autogluon.searcher import GPFIFOSearcher
import autogluon as ag

@ag.args(
    lr=ag.space.Real(1e-3, 1e-2, log=True),
    wd=ag.space.Real(1e-3, 1e-2))
def train_fn(args, reporter):
    print('lr: {}, wd: {}'.format(args.lr, args.wd))
    for e in range(10):
        dummy_accuracy = 1 - np.power(1.8, -np.random.uniform(e, 2 * e))
        reporter(epoch=e + 1, accuracy=dummy_accuracy, lr=args.lr, wd=args.wd)


print(train_fn.cs)

scheduler = ag.scheduler.FIFOScheduler(train_fn,
                                       searcher='bayesopt',
                                       resource={'num_cpus': 2, 'num_gpus': 0},
                                       num_trials=10,
                                       reward_attr='accuracy',
                                       time_attr='epoch')
scheduler.run()
```

Note that using `searcher= 'bayesopt' is equivalent to using the `GPFIFOSearcher`

**Comment by jwmueller** [comment]
@mseeger @aaronkl @tlienart  Any idea why one cannot create the GPFIFOScheduler as shown in your example:

https://autogluon.mxnet.io/api/autogluon.searcher.html

**Comment by aaronkl** [comment]
it's indeed a bug in the GPFactory, the default value for the 'scheduler' parameter is not imputed. I will send a PR to fix this

**Comment by Innixma** [comment]
Resolved in #686.
